### ISSUE TITLE
Add `_can_grab_focus` virtual function for `Control` nodes

### ISSUE THREAD

**Issue by juliohq** [godotengine/godot-proposals#6126]
### Describe the project you are working on

2D RPG game with a lot of UI programming.

### Describe the problem or limitation you are having in your project

I have to constatly set `Control.focus_mode` to either `Control.FOCUS_MODE.NONE` or `Control.FOCUS_MODE.ALL` for disabling focus for specific nodes while could be solved simply by overriding a virtual function like `_can_grab_focus`.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Add a simple virtual function that can be overriden with GDScript to check if a `Control` node can grab focus, by returning `true` or `false` accordingly.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

A very simple real life example:
```gdscript
func _can_grab_focus():
    return not disabled
```

### If this enhancement will not be used often, can it be worked around with a few lines of script?

It can't be worked around with a few lines for larger projects that rely on lot on UI.

### Is there a reason why this should be core and not an add-on in the asset library?

Other functions are already implemented like `_get_minimum_size` or `_make_custom_tooltip`.

**Comment by YuriSizov** [comment]
> I have to constatly set `Control.focus_mode` to either `Control.FOCUS_MODE.NONE` or `Control.FOCUS_MODE.ALL` for disabling focus for specific nodes while could be solved simply by overriding a virtual function like `_can_grab_focus`.

You would have to explain how a virtual function would be significantly simpler (or even much different for scripting) than setting a property.

**Comment by juliohq** [comment]
@YuriSizov added real life example and repetitive snippet of code.

**Comment by TheDuriel** [comment]
You could implement this function yourself and ducktype it.

I'm gonna go out on a limb here, and say that you don't need the ability to check _every possible_ Control for its focus ability. But only the handful of ones you are actually using, which you could easily extend to have this function.

```
class_name BetterButton
extends Button

func can_grab_focus() -> bool:
    return not Control.FOCUS_NONE == true
```

**Comment by juliohq** [comment]
@TheDuriel you're right, that would work if I had to call `can_grab_focus` manually, but this GIP is to improve how `Control` works internally, which means it needs something else besides `Control.FOCUS_MODE` to make sure a given node has focus ability and giving programmers power to decide if the node should be focusable or not should be the right thing to do.

**Comment by TheDuriel** [comment]
There is absolutely no internal improvement gained here as far as I can see.

You're replacing an if statement with a function call, and there is no scenario in which that function does anything but return the result of that very same if statement.

How about proposing changes to Control that improve usability, and require this function?

**Comment by juliohq** [comment]
@TheDuriel with a function call I can implement how I want my `Control` nodes to behave in regard to focus, instead of swiching their `focus_mode` property all the time in several places of my code. Adding a virtual function will allow the engine to "know" when a node can be focused and don't switch its stylebox "focused" stylebox (if it's a button, for example). Relying on an `if` statement to set a property isn't really the best way to do UI programming when you have a lot of `Control` nodes.

**Comment by Shadowblitz16** [comment]
do note that not all consoles nodes work with focus and focus neighbours.
even if you change the focus mode.

for example tab container completely refuses to focus its own tabs or anything from or to it
